src/nvim/api/window.c

```diff
@@ -583,3 +583,42 @@ Object nvim_win_call(Window window, LuaRef fun, Error *err)
   try_end(err);
   return res;
 }
+
+/// Gets the current viewport of a window
+///
+/// The returned tuple contains (everything is 0-based, end-exclusive):
+/// - Top-most buffer line index
+/// - Bottom-most buffer line index
+/// - Left-most column index
+/// - Right-most column index
+///
+/// @see |api-indexing|
+///
+/// @param window Window handle, or 0 for current window.
+/// @param[out] Error deailts, if any
+///
+/// @return (start_line, end_line, start_col, end_col) tuple of the viewport
+///         amount in this window.
+ArrayOf(Integer, 4) nvim_win_get_viewport(Window window, Error *err)
+  FUNC_API_SINCE(8)
```

Comment:
Maybe add a reserved `{opts}` argument in case we want to add anything in the future? Though, I'm not sure what else we could want from this.

Reply:
Great idea indeed.